### DISCUSSION TITLE
Use QUML question & questionSet API to create worksheet

### DISCUSSION THREAD

**Discussion by snehal0904** [Sunbird-inQuiry/Community#14] [General Discussions]
Hello Team,

In the Shiksha platform, we are building a usecase where teachers can create worksheets. For Shiksha, a worksheet is a collection of questions, which can be output as a PDF that the teacher can print.

For this, we intend to use the question and question set capabilities in Diksha. Here's an overview of what we want to achieve

1. Teacher begins creating a worksheet
2. Chooses questions to add. Here we wish to use the Diksha question search API to pick the questions
3. Saves the worksheet. Here we wish to use the create question set API to create a question set in Diksha.
4. Prints the worksheet. Here we intend to use the hierarchy and question read APIs, fetch all the question data, and output a PDF via Shiksha

I have 2 questions in this regard

1. We have assumed that Diksha APIs follow the QuML spec [https://github.com/sunbird-specs/QuML/blob/master/v1/question-set-schema.json](https://github.com/sunbird-specs/QuML/blob/master/v1/question-set-schema.json) However I do not see a questions attribute in the sunbird questionset API. Instead I see a childNodes which might be relevant. Wish to understand how questions are associated with a question set
2. We need a write token to be able to create question sets in Diksha, how can we get this token

Thanks!!

@Krishnaj20 @coolbung

**Comment by Krishnaj20** [comment]
@snehal0904 - Is the use case about Shiksha app fetching question sets created by teachers using the coKreat portal (Vidyadaan uses Sunbird's coKreat) or the teachers will use Shiksha app to create worksheets by fetching questions created in the cokreat portal? Please clarify.

Samagra team has worked on enabling similar use cases for one of the state tenants on DIKSHA. @Prerna789 can you assist here please?

Sunbird RC, to my knowledge, doesnt have question set schema. Can you clarify which schema are you referring to in SunbirdRC?

@pallakartheekreddy

**Reply by snehal0904** [reply]
Sorry for the confusion, please ignore the Sunbird RC related inputs.
Also, I have updated the original question to better explain the use case.

**Comment by manojSRawat** [comment]
@snehal0904

Question-set and Questions are two different entities and both have separate schema files.

Question-set: https://github.com/Sunbird-Ed/creation-portal/blob/release-4.9.0/kp_schemas/questionset/1.0/schema.json
Question: https://github.com/Sunbird-Ed/creation-portal/blob/release-4.9.0/kp_schemas/question/1.0/schema.json

We can add questions inside the question-set and this reference is stored in **childNodes**.

To play a question in a QUML player we need both a question-set and a question. 

To get both the details you can use question-set hierarchy read API. 

End Point: questionset/v1/hierarchy/do_id
Documentation: http://docs.sunbird.org/latest/apis/questionapi//#operation/Read%20QuestionSet%20Hierarchy

**Reply by pallakartheekreddy** [reply]
@snehal0904 please go through the below link for more details about Question & Questionset services
https://inquiry.sunbird.org/learn/product-and-developer-guide/question-and-question-set-service

**Reply by snehal0904** [reply]
@manojSRawat If we save in the childNodes, will it be QUML compliant?
